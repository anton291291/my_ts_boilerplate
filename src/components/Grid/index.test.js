import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';

import { Grid, Container } from './index';

describe('Grid', () => {
    test('should render correctly', () => {
        const output = shallow(<Grid />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    test('by default width prop should be 10', () => {
        const output = shallow(<Grid />);
        expect(output.prop('width')).toBe(10);
        expect(output.prop('height')).toBe(10);
    });

    test('by default it should have 100 children', () => {
        const output = shallow(<Grid />);
        expect(output.children()).toHaveLength(100);
    });

    test('by default with props = 10 , it should have width and height = 260px ', () => {
        const output = shallow(<Grid />);
        expect(output.find(Container)).toHaveStyleRule('width', '260px');
        expect(output.find(Container)).toHaveStyleRule('height', '260px');
    });
});
