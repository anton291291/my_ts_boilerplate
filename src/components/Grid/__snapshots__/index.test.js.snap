// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render correctly 1`] = `
<Grid__Container
  height={10}
  width={10}
>
  <Cell
    index={0}
    key="0"
  />
  <Cell
    index={1}
    key="1"
  />
  <Cell
    index={2}
    key="2"
  />
  <Cell
    index={3}
    key="3"
  />
  <Cell
    index={4}
    key="4"
  />
  <Cell
    index={5}
    key="5"
  />
  <Cell
    index={6}
    key="6"
  />
  <Cell
    index={7}
    key="7"
  />
  <Cell
    index={8}
    key="8"
  />
  <Cell
    index={9}
    key="9"
  />
  <Cell
    index={10}
    key="10"
  />
  <Cell
    index={11}
    key="11"
  />
  <Cell
    index={12}
    key="12"
  />
  <Cell
    index={13}
    key="13"
  />
  <Cell
    index={14}
    key="14"
  />
  <Cell
    index={15}
    key="15"
  />
  <Cell
    index={16}
    key="16"
  />
  <Cell
    index={17}
    key="17"
  />
  <Cell
    index={18}
    key="18"
  />
  <Cell
    index={19}
    key="19"
  />
  <Cell
    index={20}
    key="20"
  />
  <Cell
    index={21}
    key="21"
  />
  <Cell
    index={22}
    key="22"
  />
  <Cell
    index={23}
    key="23"
  />
  <Cell
    index={24}
    key="24"
  />
  <Cell
    index={25}
    key="25"
  />
  <Cell
    index={26}
    key="26"
  />
  <Cell
    index={27}
    key="27"
  />
  <Cell
    index={28}
    key="28"
  />
  <Cell
    index={29}
    key="29"
  />
  <Cell
    index={30}
    key="30"
  />
  <Cell
    index={31}
    key="31"
  />
  <Cell
    index={32}
    key="32"
  />
  <Cell
    index={33}
    key="33"
  />
  <Cell
    index={34}
    key="34"
  />
  <Cell
    index={35}
    key="35"
  />
  <Cell
    index={36}
    key="36"
  />
  <Cell
    index={37}
    key="37"
  />
  <Cell
    index={38}
    key="38"
  />
  <Cell
    index={39}
    key="39"
  />
  <Cell
    index={40}
    key="40"
  />
  <Cell
    index={41}
    key="41"
  />
  <Cell
    index={42}
    key="42"
  />
  <Cell
    index={43}
    key="43"
  />
  <Cell
    index={44}
    key="44"
  />
  <Cell
    index={45}
    key="45"
  />
  <Cell
    index={46}
    key="46"
  />
  <Cell
    index={47}
    key="47"
  />
  <Cell
    index={48}
    key="48"
  />
  <Cell
    index={49}
    key="49"
  />
  <Cell
    index={50}
    key="50"
  />
  <Cell
    index={51}
    key="51"
  />
  <Cell
    index={52}
    key="52"
  />
  <Cell
    index={53}
    key="53"
  />
  <Cell
    index={54}
    key="54"
  />
  <Cell
    index={55}
    key="55"
  />
  <Cell
    index={56}
    key="56"
  />
  <Cell
    index={57}
    key="57"
  />
  <Cell
    index={58}
    key="58"
  />
  <Cell
    index={59}
    key="59"
  />
  <Cell
    index={60}
    key="60"
  />
  <Cell
    index={61}
    key="61"
  />
  <Cell
    index={62}
    key="62"
  />
  <Cell
    index={63}
    key="63"
  />
  <Cell
    index={64}
    key="64"
  />
  <Cell
    index={65}
    key="65"
  />
  <Cell
    index={66}
    key="66"
  />
  <Cell
    index={67}
    key="67"
  />
  <Cell
    index={68}
    key="68"
  />
  <Cell
    index={69}
    key="69"
  />
  <Cell
    index={70}
    key="70"
  />
  <Cell
    index={71}
    key="71"
  />
  <Cell
    index={72}
    key="72"
  />
  <Cell
    index={73}
    key="73"
  />
  <Cell
    index={74}
    key="74"
  />
  <Cell
    index={75}
    key="75"
  />
  <Cell
    index={76}
    key="76"
  />
  <Cell
    index={77}
    key="77"
  />
  <Cell
    index={78}
    key="78"
  />
  <Cell
    index={79}
    key="79"
  />
  <Cell
    index={80}
    key="80"
  />
  <Cell
    index={81}
    key="81"
  />
  <Cell
    index={82}
    key="82"
  />
  <Cell
    index={83}
    key="83"
  />
  <Cell
    index={84}
    key="84"
  />
  <Cell
    index={85}
    key="85"
  />
  <Cell
    index={86}
    key="86"
  />
  <Cell
    index={87}
    key="87"
  />
  <Cell
    index={88}
    key="88"
  />
  <Cell
    index={89}
    key="89"
  />
  <Cell
    index={90}
    key="90"
  />
  <Cell
    index={91}
    key="91"
  />
  <Cell
    index={92}
    key="92"
  />
  <Cell
    index={93}
    key="93"
  />
  <Cell
    index={94}
    key="94"
  />
  <Cell
    index={95}
    key="95"
  />
  <Cell
    index={96}
    key="96"
  />
  <Cell
    index={97}
    key="97"
  />
  <Cell
    index={98}
    key="98"
  />
  <Cell
    index={99}
    key="99"
  />
</Grid__Container>
`;
